src/compas/files/gltf/data_classes.py

```diff
@@ -0,0 +1,589 @@
+        self.extensions = extensions
+
+    def to_dict(self, node_index_by_key, sampler_index_by_key):
+        channel_dict = {
+            'sampler': sampler_index_by_key[self.sampler],
+            'target': self.target.to_dict(node_index_by_key),
+        }
+        if self.extras is not None:
+            channel_dict['extras'] = self.extras
+        if self.extensions is not None:
+            channel_dict['extensions'] = self.extensions
+        return channel_dict
+
+    @classmethod
+    def from_dict(cls, channel):
+        if channel is None:
+            return None
+        return cls(
+            sampler=channel['sampler'],
+            target=TargetData.from_dict(channel['target']),
+            extras=channel.get('extras'),
+            extensions=channel.get('extensions'),
+        )
+
+
+class AnimationData(object):
+    def __init__(self, channels, samplers_dict, name=None, extras=None, extensions=None):
+        self.channels = channels
+        self.samplers_dict = samplers_dict
+        self.name = name
+        self.extras = extras
+        self.extensions = extensions
+
+        self._sampler_index_by_key = None
+
+    def to_dict(self, samplers_list, node_index_by_key):
+        channels = [
+            channel_data.to_dict(node_index_by_key, self._sampler_index_by_key)
+            for channel_data in self.channels
+        ]
+        animation_dict = {
+            'channels': channels,
+            'samplers': samplers_list,
+        }
+        if self.name is not None:
+            animation_dict['name'] = self.name
+        if self.extras is not None:
+            animation_dict['extras'] = self.extras
+        if self.extensions is not None:
+            animation_dict['extensions'] = self.extensions
+        return animation_dict
+
+    def get_sampler_index_by_key(self):
+        self._sampler_index_by_key = {key: index for index, key in enumerate(self.samplers_dict)}
+        return self._sampler_index_by_key
+
+    @classmethod
+    def from_dict(cls, animation, channel_data_list, sampler_dict):
+        if animation is None:
+            return None
+        return cls(
+            channels=channel_data_list,
+            samplers_dict=sampler_dict,
+            name=animation.get('name'),
+            extras=animation.get('extras'),
+            extensions=animation.get('extensions'),
+        )
+
+
+class SkinData(object):
+    def __init__(self, joints, inverse_bind_matrices=None, skeleton=None, name=None, extras=None, extensions=None):
+        self.joints = joints
+        self.inverse_bind_matrices = inverse_bind_matrices
+        self.skeleton = skeleton
+        self.name = name
+        self.extras = extras
+        self.extensions = extensions
+
+    def to_dict(self, node_index_by_key, accessor_index):
+        skin_dict = {'joints': [
+            node_index_by_key.get(item)
+            for item in self.joints
+            if node_index_by_key.get(item)
+        ]}
```

Comment:
I always find these multiline list comprehensions hard-ish to read. Perhaps could be split? And perhaps use `filter` instead of the duplicated `node_index_by_key.get` (although, `.get` is constant access, so, depending on the size of these lists, maybe your existing solution is better.

```suggestion
        node_indices = [node_index_by_key.get(item) for item in self.joints]
        skin_dict = {'joints': list(filter(None, node_indices))}
```

Reply:
That's the mathematician in me, finding such dictionary/list comprehensions completely natural.  Thanks for making me look at this code again because I see that there is a bug.  Since `0` is falsey, I have accidentally filtered out the zeroth node from all skins.  Since I would need a custom filter function, I think I would be most pythonic to do an amalgam of our two versions.